### DISCUSSION TITLE
Future Calls

### DISCUSSION THREAD

**Discussion by agarwalrohit2503** [serverpod/serverpod#285] [Q&A] [answered]
The given function is not defined and if not then how to do the future calls
<img width="441" alt="Screenshot 2022-09-12 at 4 39 17 PM" src="https://user-images.githubusercontent.com/33911927/189639164-d1fe1314-8e35-45dd-8110-f117a275138f.png">

**Comment by vlidholt** [comment, marked as answer]
In your `server.dart` file, you need to first register your call. E.g.

```dart
pod.registerFutureCall(ExampleFutureCall(), 'exampleFutureCall');
```

Then call it like:

```dart
session.serverpod.futureCallWithDelay(
  'exampleFutureCall',
  mySerializableObject,
  Duration(hours: 1),
);
```

**Reply by agarwalrohit2503** [reply]
OK
So there is an error in the auto generated comment  
it says call using session.server.futureCallWithDelay
![Screenshot 2022-09-12 at 4 51 40 PM](https://user-images.githubusercontent.com/33911927/189642088-cad1c4f4-e30d-4bfe-842a-9f455ea74def.png)

/lib/src/future_calls/example_future_call.dart file which gets autogenerated

If possible can i contribute to fix that?
